### PR TITLE
Feature/18199 school data layer refactor part 2

### PR DESCRIPTION
**Pull request by jon-shipley**
Refactor usages of schoolDataService.findOne to use sql methods

### CODE REVIEW

File: admin/controllers/pupil.js [hunk 1]
```diff
@@ -40,20 +32,21 @@ controller.getAddPupil = async (req, res, next, error = null) => {
...
-    'dob-day': req.body['dob-day'],
-    'dob-year': req.body['dob-year'],
-    pin: null,
-    pinExpired: false
-  }
   try {
+    const school = await schoolDataService.sqlFindOneByDfeNumber(req.user.School)
+    const pupilData = {
+      school: school.dfeNumber, // Change to `id` when saving to SQL Server
+      upn: req.body.upn && req.body.upn.trim().toUpperCase(),
```
Comment (GuyHarwood) [inline admin/controllers/pupil.js]: do we force upperCase in the service too?  feels like a good place to catch it...

File: admin/controllers/school.js [hunk 2]
```diff
@@ -3,26 +3,27 @@ const moment = require('moment')
...
+const dateService = require('../services/date.service')
 const hdfValidator = require('../lib/validator/hdf-validator')
+const headteacherDeclarationService = require('../services/headteacher-declaration.service')
+const pupilDataService = require('../services/data-access/pupil.data.service')
 const pupilService = require('../services/pupil.service')
-const pupilsNotTackingCheckService = require('../services/pupils-not-taking-check.service')
 const pupilsNotTackingCheckDataService = require('../services/data-access/pupils-not-taking-check.data.service')
-const dateService = require('../services/date.service')
-const pupilDataService = require('../services/data-access/pupil.data.service')
+const pupilsNotTackingCheckService = require('../services/pupils-not-taking-check.service')
```
Comment (GuyHarwood) [inline admin/controllers/school.js]: spelling error

File: admin/services/data-access/sql.service.js [hunk 3]
```diff
@@ -146,6 +146,9 @@ function parseResults (results) {
 /** SQL Service **/
 const sqlService = {}
 
+// Name of the admin database
+sqlService.adminDb = '[mtc_admin]'
```
Comment (GuyHarwood) [inline admin/services/data-access/sql.service.js]: this is the schema, it lives within the database.
probably just rename the property/field to `adminSchema`

File: admin/controllers/pupil.js (hunk 1 above)
Comment (jon-shipley) [reply admin/controllers/pupil.js]: Good catch!  I've moved this into the service

File: admin/controllers/school.js (hunk 2 above)
Comment (jon-shipley) [reply admin/controllers/school.js]: Fixed

File: admin/services/data-access/sql.service.js (hunk 3 above)
Comment (jon-shipley) [reply admin/services/data-access/sql.service.js]: Property now renamed to `adminSchema`

**Pull request merged**